soc/nordic/nrf54h/soc.c

```diff
@@ -82,9 +82,8 @@ static int trim_hsfll(void)
 
 static int nordicsemi_nrf54h_init(void)
 {
-#if defined(CONFIG_NRF_ENABLE_ICACHE)
```

Comment:
How about Zephyr option CONFIG_DCACHE and CONFIG_ICACHE. We won't be able to disable cache. What is more cache API is under these options and disabling them cause compilation errors